core/services/keeper/registry_interface.go

```diff
@@ -156,19 +197,19 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 			}
 		}
 
-		state, err := rw.contract1_2.GetState(opts)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get contract state at block number %d", opts.BlockNumber.Int64())
-		}
-
 		activeUpkeepIDs := make([]*big.Int, 0)
-		for int64(len(activeUpkeepIDs)) < state.State.NumUpkeeps.Int64() {
+		var activeUpkeepIDBatch []*big.Int
+		for int64(len(activeUpkeepIDs)) < upkeepCount.Int64() {
 			startIndex := int64(len(activeUpkeepIDs))
-			maxCount := state.State.NumUpkeeps.Int64() - int64(len(activeUpkeepIDs))
+			maxCount := upkeepCount.Int64() - int64(len(activeUpkeepIDs))
 			if maxCount > ActiveUpkeepIDBatchSize {
 				maxCount = ActiveUpkeepIDBatchSize
 			}
-			activeUpkeepIDBatch, err := rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(startIndex), big.NewInt(maxCount))
+			if rw.Version == RegistryVersion_1_2 {
+				activeUpkeepIDBatch, err = rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(startIndex), big.NewInt(maxCount))
+			} else {
+				activeUpkeepIDBatch, err = rw.contract1_3.GetActiveUpkeepIDs(opts, big.NewInt(startIndex), big.NewInt(maxCount))
+			}
```

Comment:
nit. can you refactor the GetActiveUpkeepIDs logic to structure this better.

One way i can think of is a separate function that takes a contract wrapper as argument, with the wrapper satisfying interface with GetState, GetActiveUpkeepIDs functions, and executes the logic within it.

It will also eliminate the need for a separate getUpkeepCount function i think

cc @EasterTheBunny may have more thoughts on how to structure this

Reply:
the problem with this is
in 1.1, we use `GetUpkeepCount` to get upkeep count,
in 1.2, we use `GetState` to get a `keeper_registry_wrapper1_**2**.GetState` object and then access the upkeep count.
in 1.3, we use `GetState` to get a `keeper_registry_wrapper1_**3**.GetState` object and then access the upkeep count. 
this makes it difficult to use a uniform interface.

similar to that, in `GetActiveUpkeepIDs`, in 1.1 we need to call `GetCanceledUpkeepList` so it cannot be in the same interface.

i feel that makes the refacotring hard. let me know what you think

Reply:
I was more thinking of a common interface for 1.2 and 1.3 since they have the same logic and same function calls, but it might be an overkill. Not a blocker imo, will leave it upto you